### PR TITLE
Easy import of GloVe vectors using Gensim

### PR DESCRIPTION
**Pull request by manasRK**
word2vec embeddings start with a line with the number of lines (tokens?) and the number of dimensions of the file. This allows gensim to allocate memory accordingly for querying the model. Larger dimensions mean larger memory is held captive. Accordingly, this line has to be inserted into the GloVe embeddings file.

### CODE REVIEW

**Comment by tmylk** [comment]
@manasRK Thanks for the PR.

It would be better to have this converter as a standalone script with usage like this:
``
python -m gensim.scripts.convert_glove_to_gensim glove_file.txt gensim_file.txt
``

If you could add a test for a small file <100Kb using [check_output](https://github.com/piskvorky/gensim/blob/develop/gensim/utils.py#L1146) that would be great

File: glove-word2vec-gensim.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+import gensim
+
+
+def prepend_line(infile, outfile, line):
+	""" 
+	Function use to prepend lines using bash utilities in Linux. 
+	(source: http://stackoverflow.com/a/10850588/610569)
+	"""
+	with open(infile, 'r') as old:
```
Comment (piskvorky) [inline glove-word2vec-gensim.py]: Use binary mode: `rb`, `wb`. Otherwise won't work on Windows.

File: glove-word2vec-gensim.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
+"""
```
Comment (piskvorky) [inline glove-word2vec-gensim.py]: Missing file header.

**Comment by piskvorky** [comment]
This PR has code style issue (indentation, spaces around binary operators etc -- see [PEP8](https://www.python.org/dev/peps/pep-0008/)).

I'm not sure I understand its purpose -- does it insert one line at the beginning of a file? Isn't easier to just use `cat`? Or what is the advantage of using this script?

**Comment by piskvorky** [comment]
Also, it seems to only work for a few hard-coded filenames (reads the dimensionality from filename?).

Isn't that too constraining / narrow purpose?

**Comment by manasRK** [comment]
Agreed @piskvorky . The major motivation to create this code was to allow easy port of GloVe vectors into Gensim. I was already using Gensim massively in my explorations with word2vec. I didn't want to write another code for GloVe separately. I found out later that there is a minor difference in the formats and hence wanted to make it Gensim-compatible. 

Not a big fan of 'hard-coding' either, but this is something that was suggested in a Github pull request earlier in my [repositories](https://github.com/manasRK/glove-gensim). My earlier approach was to calculate the` num_lines` and `dims` on the fly to create a new model file, something like this;

```
#Model File
fname="glove.6B.50d.txt" #GloVe models

#extract information to pre-pend to the final Gensim compatible file
word2vec_convert_file="word2vec_line.txt"
num_lines = sum(1 for line in open(fname))
dims= re.findall('\d+',fname.split('.')[2])

print '%d lines with %d dimensions' %(num_lines,dims)

with open(word2vec_convert_file,'wb') as f:
    f.write(str(num_lines)+ " " +str(dims) + '\n')
f.close()

model_file='glove_model.txt' #for Gensim's consumption

filenames = [word2vec_convert_file,fname]

with open(model_file, 'wb') as outfile:
    for fname in filenames:
        with open(fname) as infile:
            for line in infile:
                outfile.write(line)
outfile.close()    

```
Might be a bit in-efficient, but would love your suggestions to make it useful for guys who want to use Gensim for both GloVe and word2vec, without writing more code.

@tmylk I will add a test file soon as well. Thanks !

**Comment by piskvorky** [comment]
Sure. I think we can add such CLI script to convert glove => word2vec, it looks useful.

The script should accept the two filenames (glove, gensim) as CLI arguments, then use `smart_open` to do the actual writing/reading:

```bash
python -m gensim.scripts.glove2word2vec glove.txt word2vec.txt
```

In other words, rather than an ad-hoc script with print statements for a few hardcoded files, we definitely want this to be a generic script for any glove input.

**Comment by manasRK** [comment]
Great, I will edit my script to do the same. Make it a CLI file, which can take in arguments as you suggested and create a file to consume in Gensim.

**Comment by piskvorky** [comment]
Thanks @manasRK .

Have a look at [other scripts in the `/gensim/scripts` directory](https://github.com/piskvorky/gensim/blob/develop/gensim/scripts/word2vec_standalone.py) for inspiration and code style. Best of luck!

**Comment by manasRK** [comment]
I have edited the code for use in commandline in current form as 

`python glove2word2vec.py <GloVe vector file> <Output model file>`

Further suggestions welcome. :)

File: glove2word2vec.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2016 Manas Ranjan Kar <manasrkar91@gmail.com>
+# Licensed under the MIT License https://opensource.org/licenses/MIT
```
Comment (piskvorky) [inline glove2word2vec.py]: Gensim uses LGPL (see headers of other files).

File: glove2word2vec.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+"""
+CLI USAGE: python glove2word2vec.py <GloVe vector file> <Output model file>
+
+Convert GloVe vectors into Gensim compatible format to instantiate from an existing file on disk in the word2vec C format;
```
Comment (piskvorky) [inline glove2word2vec.py]: The format is not "gensim format"; it is the format set by the original word2vec tool (which was written in C).

File: glove2word2vec.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+model = gensim.models.Word2Vec.load_word2vec_format('/tmp/vectors.txt', binary=False)  # C text format
+
+word2vec embeddings start with a line with the number of lines (tokens?) and the number of dimensions of the file. This allows gensim to allocate memory 
+accordingly for querying the model. Larger dimensions mean larger memory is held captive. Accordingly, this line has to be inserted into the GloVe 
+embeddings file.
+
+"""
+import re
```
Comment (piskvorky) [inline glove2word2vec.py]: Empty line after module docstring.

File: glove2word2vec.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+import sys
+import gensim
+import smart_open
+
+
+def glove2word2vec(glove_vector_file,output_model_file):
```
Comment (piskvorky) [inline glove2word2vec.py]: Missing docstring.

File: glove2word2vec.py [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+    
+    def get_info(glove_file_name):
+        """
```
Comment (piskvorky) [inline glove2word2vec.py]: Docstrings in imperative (`"Calculate X..."`, `"Return Y..."` etc). See PEP8 and PEP257.

File: glove2word2vec.py [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+                new.write(str(line) + "\n")
+                for line in old:
+                    new.write(line)
+        return outfile
+        
+    
+    num_lines,dims=get_info(glove_vector_file)
+    gensim_first_line = "{} {}".format(num_lines, dims)
+    
+    print '%s lines with %s dimensions' %(num_lines,dims)
```
Comment (piskvorky) [inline glove2word2vec.py]: No print statements. Use logging to communicate progress and information to user (see other gensim files and scripts).

File: glove2word2vec.py [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+    
+    model_file=prepend_line(glove_vector_file,output_model_file,gensim_first_line)
```
Comment (piskvorky) [inline glove2word2vec.py]: Spaces around binary operators (PEP8).

File: glove2word2vec.py [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+    
+    # Demo: Loads the newly created glove_model.txt into gensim API.  
+    model=gensim.models.Word2Vec.load_word2vec_format(model_file,binary=False) #GloVe Model
```
Comment (piskvorky) [inline glove2word2vec.py]: Space after comma (PEP8).

**Comment by piskvorky** [comment]
Good progress :) I added a few more comments around code style.

**Comment by manasRK** [comment]
@piskvorky I have updated the file with your recommended changes.

File: glove2word2vec.py [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+        with smart_open.smart_open(infile, 'rb') as old:
+            with smart_open.smart_open(outfile, 'wb') as new:
+                new.write(str(line) + " \n ")
+                for line in old:
+                    new.write(line)
+        return outfile
+        
+    num_lines, dims= get_info(glove_vector_file)
+    gensim_first_line = "{} {}".format(num_lines, dims)
+    model_file=prepend_line(glove_vector_file, output_model_file, gensim_first_line)
```
Comment (piskvorky) [inline glove2word2vec.py]: Spaces around binary operator (here, above, below).

File: glove2word2vec.py [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+
+def glove2word2vec(glove_vector_file, output_model_file):
+    """Convert GloVe vectors into word2vec C format"""
+    
+    def get_info(glove_file_name):
+        """ 
+        Function to calculate the number of lines and dimensions of the GloVe vectors to make it Gensim compatible
+        """
+        num_lines = sum(1 for line in smart_open.smart_open(glove_vector_file))
+        if 'twitter' in glove_file_name:
```
Comment (piskvorky) [inline glove2word2vec.py]: We don't want to guess the dimensionality from the filename. Isn't there a way to get the dimensions from the file contents instead?

File: glove2word2vec.py [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+
+    program = sys.argv[0]
+    parser = argparse.ArgumentParser()
+    
+    parser.add_argument("-input", help="Use GloVe model file to convert into word2vec C format", type=str, required=True)
+    parser.add_argument("-output", help="Use file OUTPUT to save the resulting word vectors", type=str, default='output_vectors.txt')
+
+    logging.basicConfig(format='%(asctime)s : %(threadName)s : %(levelname)s : %(message)s', level=logging.INFO)
+    logging.root.setLevel(level=logging.INFO)
+    logger.info("running %s" % ' '.join(sys.argv))
```
Comment (piskvorky) [inline glove2word2vec.py]: Comma instead of `%` in logging.

File: glove2word2vec.py [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+    
+    args = parser.parse_args()
+
+    glove_vector_file= args.input
+    output_model_file= args.output
+
+    num_lines, dims, model, model_file= glove2word2vec(glove_vector_file, output_model_file)
+
+    logger.info('%d lines with %s dimensions', (num_lines, dims))
```
Comment (piskvorky) [inline glove2word2vec.py]: Does this work? I'd expect this fails, because info's string expects two arguments, but gets only one (a tuple of `(num_lines, dims)`).

File: glove2word2vec.py [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+    logger.info('Model %s successfully created',output_model_file)
+    
+    logger.info('Testing the model....')
+    logger.info('Most similar to king are:%s ', model.most_similar(positive=['king'], topn=10))
+    logger.info('Similarity score between woman and man is %s ' , model.similarity('woman', 'man'))
```
Comment (piskvorky) [inline glove2word2vec.py]: PEP8: no space before comma.

**Comment by piskvorky** [comment]
@manasRK a few more code style fixes -- we're getting there!

**Comment by manasRK** [comment]
Sure, let me make the fixes again. Thanks again for your detailed feedback :)

File: glove2word2vec.py (hunk 13 above)
Comment (manasRK) [inline glove2word2vec.py]: Done.

File: glove2word2vec.py (hunk 12 above)
Comment (manasRK) [inline glove2word2vec.py]: Done, I have created a function to do the following;

1. Read the first 100 lines of the embeddings file, and count number of float items
2. Appends it to a list
3. Calculate median to finalize the number of `dims`

File: glove2word2vec.py [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+        Function to calculate the number of dimensions from an embeddings file
+        """
+        count=0
+        dims=[]
+        for line in smart_open.smart_open(filename):
+            count+=1
+            if count<100:
+                dims.append(len(re.findall('[\d]+.[\d]+', line)))
+            else: break
+            return int(np.median(dims))
```
Comment (piskvorky) [inline glove2word2vec.py]: This doesn't look right. What does a "median" of dimensions mean?

Either all lines have the same number of dimensions (`len(line.split()`), in which case we only need to parse one. Or they have variable length, and then we need to take the maximum over the entire file.

**Comment by manasRK** [comment]
Done with most of the changes. A bit unsure about the PEP8 suggestion, can you help me out there?

File: glove2word2vec.py [hunk 17]
```diff
@@ -0,0 +1,100 @@
...
+    """Convert GloVe vectors into word2vec C format"""
+    
+    def count_dims(filename):
+        """ 
+        Function to calculate the number of dimensions from an embeddings file
+        """
+        count=0
+        dims=[]
+        for line in smart_open.smart_open(filename):
+            count+=1
```
Comment (piskvorky) [inline glove2word2vec.py]: PEP8: spaces around operators (here and a lot of other places).

@manasRK , does your editor support highlighting PEP8 issues? Installing a Python linter should prevent such formatting problems.

It's not a big deal, we can reformat ourselves, but this may be helpful to you in the future, when you program in Python.

File: glove2word2vec.py (hunk 16 above)
Comment (manasRK) [inline glove2word2vec.py]: I am appending a number to the list, which is the number of digits occurring in a line of the word embedding. 

The intuition is this: in a vector file, a line is a word followed by n dimension co-ordinates. I am taking the counts from first 100 lines in a list, and a median of the same to reduce chances of error.

**Comment by piskvorky** [comment]
Sure. Let me push a commit against your PR, to show what I mean.

File: glove2word2vec.py (hunk 17 above)
Comment (manasRK) [inline glove2word2vec.py]: Ah! I code in Eclipse, and have been missing them repeatedly. Will install a plugin to write better Python code from now on.

**Comment by manasRK** [comment]
Thanks @piskvorky !

**Comment by piskvorky** [comment]
Pull request sent.

**Comment by piskvorky** [comment]
@manasRK I don't have gloVe files, so I didn't run my changes at all. Can you test this PR thoroughly?

@tmylk anything else needed for the merge?

**Comment by manasRK** [comment]
image

Your approach to calculate the first line's length (51) didn't work and threw up a error (attaching screenshot).  It has some problem accepting 51 as the number of dimensions (wierd?)

I have edited the code to include the earlier approach I took, which passed on my machine upon testing giving 50 dimensions.

**Comment by piskvorky** [comment]
There's probably one extra field, which is the token itself. So it should be enough to do `- 1` in the code.

**Comment by manasRK** [comment]
Ah! You guessed it right. `-1` does the trick. Reverting the code now.

However, is there any specific reason why it should throw up an error with 51 `dims`?

**Comment by tmylk** [comment]
@manasRK is there a reason for word2vec_standalone.py being part of the PR?

Before merging please add a test for a small file <100Kb using [check_output](https://github.com/piskvorky/gensim/blob/develop/gensim/utils.py#L1146)

**Comment by manasRK** [comment]
@tmylk No, the last commit is the final version of the PR. I picked up some formatting & logging inspiration from `word2vec_standalone.py`. Will add a test file as well in sometime.

**Comment by manasRK** [comment]
@tmylk I am unable to form a test file properly, I have tested the latest commit and it works perfectly and had good amount of suggestions from @piskvorky.

Would you be able to guide me to sample codes to showcase how to use it to fit my PR? I have prepared a small sample vector file for usage as well.

File: gensim/scripts/glove2word2vec.py [hunk 18]
```diff
@@ -72,12 +72,20 @@ def glove2word2vec(glove_input_file, word2vec_output_file):
...
     logger.info('testing the model....')
-    logger.info('top-10 most similar words to "king": %s', model.most_similar(positive=['king'], topn=10))
-    logger.info('similarity score between "woman" and "man": %s', model.similarity('woman', 'man'))
+   
+    try:
+        logger.info('top-10 most similar words to "king": %s', model.most_similar(positive=['king'], topn=10))
+        logger.info('similarity score between "woman" and "man": %s', model.similarity('woman', 'man'))
+    except:
+        logger.error('"king" may not be present in the vocab of the test file. checking for model file creation now....')
+        if os.path.isfile(os.path.join(os.getcwd(), args.output)):
```
Comment (piskvorky) [inline gensim/scripts/glove2word2vec.py]: Why `getcwd`? This script has nothing to do with the "current working directory".
Use `args.output` only.

File: gensim/scripts/glove2word2vec.py [hunk 19]
```diff
@@ -72,12 +72,20 @@ def glove2word2vec(glove_input_file, word2vec_output_file):
...
+            logger.info('model file %s successfully created in directory %s. pLease check with other words/phrases', args.output, os.getcwd())
```
Comment (piskvorky) [inline gensim/scripts/glove2word2vec.py]: Typo: `pLease` => `Please`.

Also, just pick a (random? first? last?) word from the original gloVe model to test. No need to instruct users to pick it themselves.

Comment (manasRK) [inline gensim/scripts/glove2word2vec.py]: Done.

File: gensim/scripts/glove2word2vec.py [hunk 20]
```diff
@@ -0,0 +1,96 @@
...
+    
+    # test that the converted model loads successfully
+    model = gensim.models.Word2Vec.load_word2vec_format(args.output, binary=False)
+    logger.info('model %s successfully loaded', model)
+    
+   
+    try:
+        logger.info('testing the model....')
+        with smart_open(args.output, 'rb') as f:
+            seed_word1, seed_word2 = random.sample([each.split()[0] for each in f.readlines()], 2)
```
Comment (piskvorky) [inline gensim/scripts/glove2word2vec.py]: `readlines()` unnecessary (the Python pattern is `for line in f`).

File: gensim/scripts/glove2word2vec.py [hunk 21]
```diff
@@ -0,0 +1,96 @@
...
+    
+    # do the actual conversion
+    num_lines, num_dims = glove2word2vec(args.input, args.output)
+    logger.info('converted model with %i vectors and %i dimensions', num_lines, num_dims)
+    
+    # test that the converted model loads successfully
+    model = gensim.models.Word2Vec.load_word2vec_format(args.output, binary=False)
+    logger.info('model %s successfully loaded', model)
+    
+
```
Comment (piskvorky) [inline gensim/scripts/glove2word2vec.py]: PEP8: too many blank lines.

File: gensim/scripts/glove2word2vec.py [hunk 22]
```diff
@@ -0,0 +1,96 @@
...
+    model = gensim.models.Word2Vec.load_word2vec_format(args.output, binary=False)
+    logger.info('model %s successfully loaded', model)
+    
+   
+    try:
+        logger.info('testing the model....')
+        with smart_open(args.output, 'rb') as f:
+            seed_word1, seed_word2 = random.sample([each.split()[0] for each in f.readlines()], 2)
+            
+        logger.info('top-10 most similar words to %s: %s', seed_word1, model.most_similar(positive=['the'], topn=10))
```
Comment (piskvorky) [inline gensim/scripts/glove2word2vec.py]: Hardcoded query as `'the'`.

File: gensim/scripts/glove2word2vec.py [hunk 23]
```diff
@@ -0,0 +1,96 @@
...
+        logger.info('similarity score between %s and %s: %s', seed_word1, seed_word2, model.similarity(seed_word1, seed_word2))
+    except:
+        logger.error(' error encountered. checking for model file creation now....', seed_word1)
```
Comment (piskvorky) [inline gensim/scripts/glove2word2vec.py]: Argument without a matching format string.

Comment (manasRK) [inline gensim/scripts/glove2word2vec.py]: Made all the above recommended changes.

**Comment by piskvorky** [comment]
This looks like a sisyphian task @manasRK! Please go and read PEP8 and PEP257 before making further code style changes.

**Comment by manasRK** [comment]
@piskvorky I have used the python packages to be sure of the errors I am making for PEP8 & PEP257 standards. I have never paid much attention to formatting till now, will hope that I have been able to correct it now.

**Lesson learnt**: Formatting is key.

**Comment by manasRK** [comment]
@tmylk I have added a test file and the test data as well. Would you need anything else?

Currently the check is failing as it is unable to find glove2word2vec.py in gensim.scripts, though I am able to run the programs without any unexpected behavior on my machine.

**Comment by piskvorky** [comment]
Thanks.

CLI scripts in Python libraries are run using the `-m` parameter: `python -m gensim.scripts.glove2word2vec -i x -o y`. That way, the script can be launched from anywhere, as long as the library is installed.

**Comment by manasRK** [comment]
Thanks @piskvorky . Changes done.

**Comment by manasRK** [comment]
@piskvorky @tmylk 

- Build for Python >3 fails citing `AttributeError: '_io.BufferedReader' object has no attribute 'next'`
  from the code snippet for dims `num_dims = len(smart_open(glove_file_name).next().split()) - 1`

- Build for Python 2.6 fails citing `Import Error : No module named argparse`

Any suggestions?

**Comment by piskvorky** [comment]
`argparse` is fine (we can use a backport on 2.6); the fix for Python 3 shouldn't be difficult either.

@tmylk can you take over from here? I think the PR is in good shape now, with tests too. Thanks a lot @manasRK !

**Comment by manasRK** [comment]
No ! Thank you @piskvorky for being ultra-patient with my incessant pings and formatting errors. My next PR (whenever that is) should be much less painful. :)

**Comment by manasRK** [comment]
@tmylk Please let me know if there is anything else needs to be done at my end on this. Thanks !

**Comment by piskvorky** [comment]
@tmylk ping

**Comment by tmylk** [comment]
@manasRK 
The suggestions and fixes are in https://github.com/piskvorky/gensim/pull/634  The unit tests are now in their own class now as well.

Please review and update this PR.

There are still string handling errors in Python 2.6 and 3. If you could fix them it would be great.

**Comment by piskvorky** [comment]
@tmylk why not submit the PR against @manasRK branch, so he can review/merge?

Or what is the purpose of PR #634 inside piskvorky/gensim?

**Comment by tmylk** [comment]
@manasRK didn't realise you opened up your fork for PRs. Usually a request to be added as collaborator takes some time. Thanks for thinking ahead.
I will submit next one as PR, though it looks like you are fixing the issues yourself. Ping me here when you think you are at a good place. (BTW you can also setup Travis to test from your fork.)

**Comment by piskvorky** [comment]
@tmylk All public repos on github are open to PRs, no special action needed.

**Comment by manasRK** [comment]
@tmylk Finally managed to get rid of string literal errors with Python 3. And I officially hate Python 3 now. :(

Helpful link for anyone who follows this thread later, difference between Python 3 and Python 2.

http://lucumr.pocoo.org/2014/1/5/unicode-in-2-and-3/

**Comment by tmylk** [comment]
Great job. Thanks a lot. Did you run pep8 on it?

**Comment by manasRK** [comment]
Yes, checked with PEP8 and PEP257 python packages as well.

File: setup.py [hunk 24]
```diff
@@ -114,6 +104,11 @@ def readfile(fname):
     import wheelhouse_uploader.cmd
     cmdclass.update(vars(wheelhouse_uploader.cmd))
 
+python_2_6_backports = ''
```
Comment (piskvorky) [inline setup.py]: What does an "empty string" inside setup's `install_requires` do? Is that ok?

File: setup.py [hunk 25]
```diff
@@ -40,24 +38,16 @@ class custom_build_ext(build_ext):
 be compiled. No C extensions are essential for gensim to run,
 although they do result in significant speed improvements for some modules.
 %s
-
 Here are some hints for popular operating systems:
-
 If you are seeing this message on Linux you probably need to
 install GCC and/or the Python development package for your
 version of Python.
-
```
Comment (piskvorky) [inline setup.py]: We want to keep all these paragraph breaks. Why remove them? It makes the text hard to follow.

Comment (manasRK) [inline setup.py]: Done.

File: setup.py (hunk 24 above)
Comment (manasRK) [inline setup.py]: This comes from #634 . @tmylk can shed some light on this.

File: setup.py [hunk 26]
```diff
@@ -6,7 +6,6 @@
 
 """
 Run with:
-
```
Comment (piskvorky) [inline setup.py]: @manasRK I still see many such deleted lines in the diff (top tab, "Files changed").

File: setup.py (hunk 24 above)
Comment (tmylk) [inline setup.py]: We install argparse only if Python version is 2.6. By default we don't need to install any backports, hence an empty string.

File: setup.py (hunk 26 above)
Comment (manasRK) [inline setup.py]: Re-added deleted blank lines.

File: setup.py (hunk 24 above)
Comment (piskvorky) [inline setup.py]: Question was, what does an empty string imply in `install_requires`. Ignored? Warning? Exception?

Comment (tmylk) [inline setup.py]: The empty string doesn't cause any output in Travis.
https://travis-ci.org/piskvorky/gensim/jobs/118421963#L290

**Comment by manasRK** [comment]
@tmylk @piskvorky Codes are now okay? Will close the PR if you are ready to merge :)

**Comment by piskvorky** [comment]
Yes, this is in @tmylk 's hands now, thanks @manasRK !

File: gensim/scripts/glove2word2vec.py [hunk 27]
```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2016 Radim Rehurek <radimrehurek@seznam.cz>
+# Copyright (C) 2016 Manas Ranjan Kar <manasrkar91@gmail.com>
```
Comment (tmylk) [inline gensim/scripts/glove2word2vec.py]: Just checking that you are aware of the legal guidelines of contributing. "By submitting your contribution to be included in the gensim project, you agree to assign joint ownership of your changes (your code patch, documentation fix, whatever) to me, Radim Řehůřek."

https://github.com/piskvorky/gensim/wiki/Developer-page#legal

Comment (manasRK) [inline gensim/scripts/glove2word2vec.py]: Yes, surely. That's why I added @piskvorky 's name in the code.

File: gensim/scripts/word2vec_standalone.py [hunk 28]
```diff
@@ -10,8 +10,7 @@
 -min_count MIN-COUNT -alpha ALPHA -binary BINARY -accuracy FILE
```
Comment (tmylk) [inline gensim/scripts/word2vec_standalone.py]: Please remove this file from this PR.

**Comment by tmylk** [comment]
Please add note about this new feature to changelog file.
Also see comments about copyright and removing word2vec_standalone.

If you resubmit the PR against the develop branch then it will be merged in.

Master is always the latest pypi release.

**Comment by piskvorky** [comment]
My notes:

Doc fixes in `word2vec_standalone` are fine. Joint copyright ownership also fine.

If the PR against `develop` branch or squashing the commits gives you trouble @manasRK, @tmylk can help you with this. Let's get this merged ASAP.

**Comment by tmylk** [comment]
@manasRK Happy to talk in https://gitter.im/piskvorky/gensim if needed

**Comment by tmylk** [comment]
Merged in https://github.com/piskvorky/gensim/commit/982dc3c39f39479084072456f54a4feb2fcdc068
